webdav/prop.go

```diff
@@ -412,6 +416,36 @@ func findDisplayName(ctx context.Context, fs FileSystem, ls LockSystem, name str
 	return escapeXML(fi.Name()), nil
 }
 
+func findLockDiscovery(ctx context.Context, fs FileSystem, ls LockSystem, name string, fi os.FileInfo) (string, error) {
+	if slashClean(name) == "/" {
+		// Hide the real name of a possibly prefixed root directory.
+		return "", nil
+	}
+
+	lockDetails, err := ls.LockDetails(name, fi)
+	if err != nil || lockDetails == nil {
+		return "", nil
+	}
+
+	detailsXML := fmt.Sprintf("<D:activelock>"+
+		"<D:locktype><D:write/></D:locktype>"+
+		"<D:lockscope><D:exclusive/></D:lockscope>"+
+		"<D:owner>%s</D:owner>",
+		lockDetails.OwnerXML,
+	)
+
+	if lockDetails.Duration.Seconds() < 0 {
+		detailsXML = detailsXML + "<D:timeout>Infinite</D:timeout>"
```

Comment:
We could probably do this in the Sprintf above. Would likely be more efficient than concatenation.